templates/demo-store/app/root.tsx

```diff
@@ -97,11 +97,20 @@ export default function App() {
   );
 }
 
-export function CatchBoundary() {
+// Can't find this type exported anywhere
```

Comment:
Is it not automatically typed as the return type of `useRouteError()`?

If so, I would raise an issue with Remix on this.

Reply:
You can use type narrowing via `isRouteErrorResponse()` to get type safety on these.  I think originally we didn't export it directly since its a class and we don't want folks doing `new ErrorResponse`